### PR TITLE
add --add-host argument to agents

### PR DESCRIPTION
**Pull request by mattnworb**
This argument allows for specifying entries to be added to the
`/etc/hosts` file of the containers created by the Agent, mirroring the
functionality of the [`--add-host` argument to `docker run`][docker] or `docker
create`.

It is implemented within the Agent by passing the `host:ip` pairs in the
`HostConfig.ExtraHosts` field when [creating the container][remote-api].

[docker]: https://docs.docker.com/engine/reference/commandline/run/#add-host-device-to-container-device
[remote-api]: https://docs.docker.com/engine/reference/api/docker_remote_api_v1.18/#create-a-container

### CODE REVIEW

File: helios-services/src/main/java/com/spotify/helios/agent/AgentParser.java [hunk 1]
```diff
@@ -149,25 +152,37 @@ public AgentParser(final String... args) throws ArgumentParserException {
...
+
+    agentConfig.setBinds(validateArgument(
+        options.getList(bindArg.getDest()),
+        BindVolumeContainerDecorator::isValidBind,
+        arg -> "Invalid bind " + arg));
+
+    agentConfig.setExtraHosts(validateArgument(
+        options.getList(addHostArg.getDest()),
+        AddExtraHostContainerDecorator::isValidArg,
+        arg -> "Invalid ExtraHost " + arg));
```
Comment (mattnworb) [inline helios-services/src/main/java/com/spotify/helios/agent/AgentParser.java]: please tell me if this and the `validateArgument` function below is way too much functional-style

Comment (davidxia) [inline helios-services/src/main/java/com/spotify/helios/agent/AgentParser.java]: Nice, I like it.

File: helios-services/src/main/java/com/spotify/helios/agent/AgentParser.java [hunk 2]
```diff
@@ -149,25 +152,37 @@ public AgentParser(final String... args) throws ArgumentParserException {
...
+
+  /**
+   * Verifies that all entries in the Collection satisfy the predicate. If any do not, throw an
+   * IllegalArgumentException with the specified message for the first invalid entry.
+   */
+  private static <T> List<T> validateArgument(List<T> list, Predicate<T> predicate,
+                                              Function<T, String> msgFn) {
+
+    final Optional<T> firstInvalid = list.stream()
+        .filter(predicate)
```
Comment (davidxia) [inline helios-services/src/main/java/com/spotify/helios/agent/AgentParser.java]: Do you mean `predicate.negate()`?

Comment (mattnworb) [inline helios-services/src/main/java/com/spotify/helios/agent/AgentParser.java]: image

File: helios-services/src/main/java/com/spotify/helios/agent/AddExtraHostContainerDecorator.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2014 Spotify AB.
```
Comment (gimaker) [inline helios-services/src/main/java/com/spotify/helios/agent/AddExtraHostContainerDecorator.java]: nitpick: it's no longer 2014 :)

File: helios-services/src/test/java/com/spotify/helios/agent/AddExtraHostContainerDecoratorTest.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2014 Spotify AB.
```
Comment (gimaker) [inline helios-services/src/test/java/com/spotify/helios/agent/AddExtraHostContainerDecoratorTest.java]: nitpick: 2014 -> 2016

**Comment by gimaker** [comment]
Relevant tests are failure, otherwise LGTM

**Comment by mattnworb** [comment]
Tests are green now !

**Comment by mattnworb** [comment]
@spotify/helios-team

**Comment by negz** [comment]
LGTM

**Pull request merged**